packages/insomnia/src/ui/routes/remote-collections.tsx

```diff
@@ -341,6 +339,13 @@ export const syncDataLoader: LoaderFunction = async ({
     remoteBranchesCache[workspaceId] = remoteBranches;
     remoteCompareCache[workspaceId] = compare;
     remoteBackendProjectsCache[workspaceId] = remoteBackendProjects;
+    // update workspace meta with sync data, use for show unpushed changes on collection card
+    models.workspaceMeta.updateByParentId(workspaceId, {
+      syncData: {
+        hasUncommittedChanges: Object.keys(status?.unstaged || {}).length > 0 || Object.keys(status?.stage || {}).length > 0,
+        hasUnpushedChanges: compare?.ahead > 0,
+      },
+    });
```

Comment:
Thinking this is simpler to understand. What do you think? Feel free to add/dismiss this one:
```suggestion
    let hasUncommittedChanges = false;
    if (status?.unstaged && Object.keys(status.unstaged).length > 0) {
      hasUncommittedChanges = true;
    }
     if (status?.stage && Object.keys(stage).length > 0) {
      hasUncommittedChanges = true;
    }
    // update workspace meta with sync data, use for show unpushed changes on collection card
    models.workspaceMeta.updateByParentId(workspaceId, {
      syncData: {
        hasUncommittedChanges,
        hasUnpushedChanges: compare?.ahead > 0,
      },
    });
```

Reply:
Good advice. I agree with you👍